compass/ocean/tests/global_ocean/init/initial_state.py

```diff
@@ -160,6 +170,36 @@ def run(self):
 
         run_model(self)
 
+        if self.with_inactive_top_cells:
+
+            logger.info("   * Updating minLevelCell for inactive top cells")
+
+            in_filename = 'initial_state.nc'
+            out_filename = in_filename
+
+            with xarray.open_dataset(in_filename) as ds:
+
+                # keep the data set with Time for output
+                ds_out = ds
+
+                ds = ds.isel(Time=0)
+
+                if ('minLevelCell' in ds):
+                    if config.has_option('vertical_grid', 
+                                         'inactive_top_cells'):
+                        offset = config.getint('vertical_grid', 
+                                               'inactive_top_cells')
+                    minLevelCell = ds.minLevelCell+offset
+                    ds_out['minLevelCell'] = minLevelCell
+                else:
+                    logger.info("   - Streams missing for inactive top cells")
+
+            remove_inactive_top_cells_output(in_filename, 
+                                             inactive_top_cells=offset)
```

Comment:
```suggestion
                if config.has_option('vertical_grid', 'inactive_top_cells'):
                    offset = config.getint('vertical_grid',
                                           'inactive_top_cells')
                else:
                    offset = 0

                if 'minLevelCell' in ds:
                    minLevelCell = ds.minLevelCell+offset
                    ds_out['minLevelCell'] = minLevelCell
                else:
                    logger.info("   - Variable minLevelCell, needed for "
                                "inactive top cells, is missing from the "
                                "initial condition")

            remove_inactive_top_cells_output(in_filename, 
                                             inactive_top_cells=offset)
```
The logic here needs to be rearranged so that `offset` is always defined since you use it int the call to `remove_inactive_top_cells_output()`.  I think a more verbose output is needed as I have suggested.  But I actually think you should have this raise a `ValueError` instead of just printing a log message.  If `minLevelCell` is missing, something has gone wrong.